File: java/matchsticks.java
import java.util.Scanner;

public class matchsticks {
	static String[] smallest = new String[99];
	static {
		smallest[0] = "1";
		smallest[1] = "7";
		smallest[2] = "4";
		smallest[3] = "2";
		smallest[4] = "6";
		smallest[5] = "8";
		smallest[6] = "10";
		smallest[7] = "18";
		smallest[8] = "22";
		smallest[9] = "20";
		smallest[10] = "28";
		smallest[11] = "68";
		smallest[12] = "88";
		smallest[13] = "108";
		smallest[14] = "188";
		smallest[15] = "200";
		for (int i = 16; i < smallest.length; i++)
			smallest[i] = smallest[i - 7] + "8";
	}

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int T = scan.nextInt();
		while (T-- > 0) {
			int number = scan.nextInt();
			int length = number / 2;
			StringBuilder largest = new StringBuilder(length);
			for (int i = 0; i < length; i++)
				largest.append("1");
			if (number % 2 != 0)
				largest.setCharAt(0, '7');
			System.out.println(smallest[number - 2] + " " + largest);
		}
		scan.close();
	}
}

File: ts/src/matchsticks.ts
import readline = require('readline')

function generateSmallest (): string[] {
  const smallest = [
    '1',
    '7',
    '4',
    '2',
    '6',
    '8',
    '10',
    '18',
    '22',
    '20',
    '28',
    '68',
    '88',
    '108',
    '188',
    '200'
  ]
  for (let i = 16; i < 99; i++) {
    smallest[i] = smallest[i - 7] + '8'
  }
  return smallest
}

function getMatchSticks (num: number): [string, string] {
  const length = Math.floor(num / 2)
  const largest: string[] = []
  for (let i = 0; i < length; i++) {
    largest.push('1')
  }
  if (num % 2 !== 0) {
    largest[0] = '7'
  }
  return [smallests[num - 2] ?? '', largest.join('')]
}

let lineCount = 0
const smallests: string[] = generateSmallest()

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    if (lineCount > 0) {
      console.log(getMatchSticks(parseInt(line)).join(' '))
    }
    lineCount++
  })
  .on('close', () => {
    process.exit(0)
  })
